### PR TITLE
net/tcp: correct the port byte order 

### PR DESCRIPTION
**Pull request by anchao**
## Summary

net/tcp: correct the port byte order 

1. correct the port byte order
2. Do not re-select the port if the local port has been bind()

Signed-off-by: chao.an <anchao@xiaomi.com>

## Impact

tcp connect()/bind()

## Testing

socketpair SOCK_STREAM  stress testing

### CODE REVIEW

File: net/tcp/tcp_conn.c [hunk 1]
```diff
@@ -1133,42 +1133,49 @@ int tcp_connect(FAR struct tcp_conn_s *conn, FAR const struct sockaddr *addr)
...
+        {
+          /* Select a port that is unique for this IPv4 local address (host
+           * order).
+           */
 
-      port = tcp_selectport(PF_INET,
-                            (FAR const union ip_addr_u *)&conn->u.ipv4.laddr,
-                            ntohs(conn->lport));
-    }
+          port = tcp_selectport(PF_INET,
```
Comment (antmerlino) [inline net/tcp/tcp_conn.c:1151]: Need to use htons() so that `port` is in network order for assignment to lport below.

File: net/tcp/tcp_conn.c [hunk 2]
```diff
@@ -1133,42 +1133,49 @@ int tcp_connect(FAR struct tcp_conn_s *conn, FAR const struct sockaddr *addr)
...
+        {
+          /* Select a port that is unique for this IPv6 local address (host
+           * order).
+           */
 
-      port = tcp_selectport(PF_INET6,
-                            (FAR const union ip_addr_u *)conn->u.ipv6.laddr,
-                            ntohs(conn->lport));
-    }
+          port = tcp_selectport(PF_INET6,
```
Comment (antmerlino) [inline net/tcp/tcp_conn.c:1166]: same as above

**Review by antmerlino — CHANGES_REQUESTED** [review]

File: net/tcp/tcp_conn.c [hunk 3]
```diff
@@ -285,19 +286,18 @@ static int tcp_selectport(uint8_t domain, FAR const union ip_addr_u *ipaddr,
...
-
-          /* Make sure that the port number is within range */
-
-          if (g_last_tcp_port >= 32000)
+          if (++g_last_tcp_port >= 32000)
             {
               g_last_tcp_port = 4096;
             }
+
+          portno = htons(g_last_tcp_port);
```
[the remark this answers is not in the document]
Comment (anchao) [reply net/tcp/tcp_conn.c:298]: Hi @antmerlino ,

Thanks for review, I update the change and unifies the byte orders to network, please review again

File: net/tcp/tcp_conn.c (hunk 1 above)
Comment (anchao) [reply net/tcp/tcp_conn.c:1151]: Done

**Review by antmerlino — APPROVED** [review]
Changes now look good @anchao

**Pull request merged**